hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -750,6 +751,25 @@ public void renameKey(String volumeName, String bucketName,
     ozoneManagerClient.renameKey(keyArgs, toKeyName);
   }
 
+  @Override
+  public void renameKeys(String volumeName, String bucketName,
+                         Map<String, String> keyMap) throws IOException {
+    verifyVolumeName(volumeName);
+    verifyBucketName(bucketName);
+    HddsClientUtils.checkNotNull(keyMap);
+    Map <OmKeyArgs, String> keyArgsMap = new HashMap<>();
+    for (Map.Entry< String, String > entry : keyMap.entrySet()) {
+      OmKeyArgs keyArgs = new OmKeyArgs.Builder()
+          .setVolumeName(volumeName)
+          .setBucketName(bucketName)
+          .setKeyName(entry.getKey())
+          .build();
+      keyArgsMap.put(keyArgs, entry.getValue());
+    }
```

Comment:
What's the benefit of building a `Map<OmKeyArgs, String>` compared to passing plain `Map<String, String> keyMap` directly?

Reply:
The main purpose here is to keep up with the [renameKey implementation](https://github.com/apache/hadoop-ozone/pull/1150/files#diff-ddfee0637dda7038422139fc5a3cbfdeR751) and help make the code implementation clearer.

Reply:
The signature of `OzoneManagerProtocol#renameKeys(Map<OmKeyArgs, String> keyMap)` accepts a batch of keys in different volumes/buckets.  But `RpcClient#renameKeys` only supports renaming multiple keys in a single volume/bucket.

My concern is:

1. lots of object conversion `String -> OmKeyArgs -> KeyArgs`
2. volume/bucket existence check performed for each key

I think given the limitation in current `RpcClient` the above is unnecessary overhead.

Reply:
Here I'll modify RPCClient to use Map <OmKeyArgs, String> keyArgsMap directly.

Reply:
It seems none of the methods implementing `ClientProtocol` have `OmKeyArgs` in the signature.  So I don't think `renameKeys` should do that either.

Reply:
Thanks for @adoroszlai 's reply. At present, our client calls RpcClient#renameKeys through [OzoneBucket#renameKeys](https://github.com/apache/hadoop-ozone/pull/1150/files#diff-a4a0be1d5df61f0de3bdce909bb3a4a3R1273) to batch renameKeys.  OzoneBucket is a bucket-level API, we can only rename Keys in a single volume/bucket use it.
Do we need to change the RpcClient#renameKeys? If need change here any Suggestions here?

Reply:
Thanks @captainzmc for removing the existence check.  However, bucket link resolution is still performed for each key, which seems unnecessary since all keys are in the same volume/bucket.

Reply:
Thanks @adoroszlai's feedback.  I had added new commit to fix the problem.

Reply:
I agree with @adoroszlai here, we don't need KeyArgs.

HDDS-3930 created a new message DeleteKeyArgs {
    required string volumeName = 1;
    required string bucketName = 2;
    repeated string keys = 3;
}

we can come up with a new proto structure here.


message RenameKeysRequest {
    optional RenameKeyArgs deleteKeys = 1;
}

message RenameKeyArgs {
    required string volumeName = 1;
    required string bucketName = 2;
    repeated string keys = 3;
}

I am not completely sure what is the benefit to be consistent with RenameKeyRequest message structure.

Reply:
sorry missed this, it has been already addressed. But I see we pass volumeName/bucketName with each Renamekey, we can set them only once, as right now rename is supported within a single volume/bucket

Reply:
Thanks for @bharatviswa504 's suggestion.  Agree with your idea, I had already updated this PR. KeyArgs is no longer used. And set volumeName and bucketName only once. Can you help take another look?